### ISSUE TITLE
SFTP Upload to Ubuntu server sub directory

### ISSUE THREAD

**Issue by g0Idfinger** [Velocidex/velociraptor#1502]
Running following command:
`Velociraptor.exe" query -v "SELECT upload_sftp(file='test.txt', path='test', user='sftp', privatekey=read_file(filename='sftp.key'), endpoint='x.x.x.x:22') FROM scope()"`

Result is file `test.txt` being name `'test/test.txt'` on the `/` of the ftp server.  This is a filename and not a directory path and file name.

Expected to be uploaded to directory `/test/test.txt`

I am NOT uploading to an S3 bucket but my own Ubuntu server.  I want to be able to upload to different folders based on criteria.

**Comment by scudette** [comment]
This works for me:
```sql

SELECT upload_sftp(
  endpoint="127.0.0.1:22", user="sftpuser", 
  file="/tmp/hello.txt", path="/dir2/",  name="a1.txt",
  privatekey=read_file(filename="/tmp/sftpuser_id_rsa.pem")),
  read_file(filename="/tmp/hello.txt")
 FROM scope()
```

![Velociraptor_Response_and_Monitoring](https://user-images.githubusercontent.com/3856546/150185837-243485f4-6fc5-4b63-bf2e-95dc9698ce59.png)

For reference this is how the SFTP server is configured - in /etc/ssh/sshd_config

```
Match User sftpuser
    ForceCommand internal-sftp
    PasswordAuthentication yes
    ChrootDirectory /var/sftp
    PermitTunnel no
    AllowAgentForwarding no
    AllowTcpForwarding no
    X11Forwarding no
```

**Comment by g0Idfinger** [comment]
when you `ls` that upload directory the file is located in the directory `dir2`?  when I try it, it will prepend the 'path' to the filename and put it on the / of the ftp.  
![image](https://user-images.githubusercontent.com/60414807/150187437-48acfed4-e2e7-41ec-9644-a65bc1bbeffe.png)
as you can see in this screen shot the folder 'test' is in blue and the files I tried to put into test are just files in green.

**Comment by scudette** [comment]
Yeah it does the right thing

![screen](https://user-images.githubusercontent.com/3856546/150188019-b1ef3e7f-ecf9-4bb5-b3cd-4c1f7bcfd26b.png)

As I mentioned before I think this might be because the version you are using has this bug where it uses \ as a path separator which might be confusing sftp - try 0.6.3 

https://github.com/Velocidex/velociraptor/blob/11145fd469e9ee193dca01c61fe4a3faecae81e8/vql/tools/sftp_upload.go#L192

Here it uses filepath which uses \ on windows and / on linux. 

If you run the same thing on windows with 0.6.2 you get a permission denied 
![Windows_Test_VM_2021](https://user-images.githubusercontent.com/3856546/150190422-debdd397-05fc-4980-ab8e-40b7fb88287d.png)

but this is fixed in 0.6.3
![Windows_Test_VM_2021](https://user-images.githubusercontent.com/3856546/150190761-9e653e9b-9ca3-4db2-82d4-19246c63d107.png)

**Comment by g0Idfinger** [comment]
Thank you for your helping me find my mistake, server is 6.3 but the exe was 6.2 my server build pulled down the wrong exe.  I have tested and it is working as you say it should.  Keep up the great work!
